src/poliastro/tests/test_iod.py

```diff
@@ -17,12 +20,21 @@ def test_vallado75(lambert):
     expected_va = [2.058925, 2.915956, 0.0] * u.km / u.s
     expected_vb = [-3.451569, 0.910301, 0.0] * u.km / u.s
 
-    va, vb = next(lambert(k, r0, r, tof))
+    (va, vb) = next(lambert(k, r0, r, tof))
     assert_quantity_allclose(va, expected_va, rtol=1e-5)
     assert_quantity_allclose(vb, expected_vb, rtol=1e-4)
 
+#Added by Riot: @JonPsy Github : @NanuSai
+@pytest.mark.parametrize('lambert',[vallado.lambert])
+def test_issue840(lambert):
+    k = (c.GM_earth.to(u.km ** 3/ u.s ** 2))
+    r0 = [10000., 0, 0] * u.km
+    rf = [8000., -5000, 0] * u.km
+    tof = 2 * u.hour
+    assert(next(lambert(k,r0,rf,tof,short=False)) != None)
 
```

Comment:
Running this test with the master branch would through Runtime error "Maximum error reached" and hence would fail the test.  Running it on my PR would return a generator object and hence would pass the test 😊